server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/InternalComposite.java

```diff
@@ -528,7 +529,7 @@ public int compareTo(ArrayMap that) {
         }
     }
 
-    private static int compareNullables(Comparable a, Comparable b) {
+    private static <T extends Comparable<T>> int compareNullables(T a, T b) {
```

Comment:
Shouldn't this be `T extends Comparable<?>>`?

Reply:
Or couldn't we just have made the method arguments `Comparable<?>`?

Reply:
Neither of those options works in practice, because one `Comparable<?>` isn't necessarily the same type as another - javac just can't know.

I think the type signature is correct - the method compares two possibly-null comparables of the same type. It does mean there's some regrettable raw types on line 515. I could change the arguments to just `Comparable`, but that just moves the suppression to this method instead. I'm not sure there's a good option.

Reply:
Ah, I see what's happening here. Yep, gotcha. This makes sense now.